test/parallel/test-repl-persistent-history.js

```diff
@@ -31,21 +30,19 @@ class ActionStream extends stream.Stream {
       const next = _iter.next();
       if (next.done) {
         // Close the repl. Note that it must have a clean prompt to do so.
-        setImmediate(() => {
-          this.emit('keypress', '', { ctrl: true, name: 'd' });
-        });
+        this.emit('keypress', '', { ctrl: true, name: 'd' });
```

Comment:
I guess keeping `setImmediate()` could be kept? I would rather keep it to pretend the operations are handled async.
The same applies to the other places.

Reply:
I did it this way because all the other tests with similar structure don't have the setImmediate, I did it for consistency.  
Happy to keep it here tho, I don't have strong opinions about this

Reply:
I think it might be nice to have some that effectively work async. That way we check for sync and async behavior properly.